src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,72 @@
+import cn from 'classnames';
+
+enum FilterQuery {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
+
+type Props = {
+  filteredItemsCount: number
+  onFilterChange: (value: FilterQuery) => void
+  filterQuery: FilterQuery
+};
+
+export const Footer:React.FC<Props> = ({
+  filteredItemsCount,
+  onFilterChange,
+  filterQuery,
+}) => {
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="todosCounter">
+        {`${filteredItemsCount} items left`}
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          data-cy="FilterLinkAll"
+          href="#/"
+          className={cn('filter__link',
+            { selected: filterQuery === FilterQuery.All })}
+          onClick={() => {
+            onFilterChange(FilterQuery.All);
```

Comment:
because of using `onFilterChange` it looks totally unclear. If you leave it just `setFilter` or sth - I will know exactly what is it about